File: AdityaKumar1010707/PrefixSum & SuffixSum/minimumValueToGetStepByStepSum.cpp
#include <iostream>
#include <vector>
#include <limits.h>
#include <cmath>

using namespace std;

class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int minPrefixSum = INT_MAX;
        int runningSum = 0;

        for (int num : nums) {
            runningSum += num;
            minPrefixSum = min(minPrefixSum, runningSum);
        }

        return (minPrefixSum < 0) ? abs(minPrefixSum) + 1 : 1;
    }
};

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;

    vector<int> nums(n);
    cout << "Enter " << n << " numbers: ";
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    Solution obj;
    int result = obj.minStartValue(nums);

    cout << "Minimum start value required: " << result << endl;

    return 0;
}

File: [Manya]/Leetcode/Prefix/Minimum_Value_to_get_step_by_step_sum.java
import java.util.Scanner;

class Solution {
    public int minStartValue(int[] nums) {
        int minSum = 0;
        int currSum = 0;

        for(int num : nums){
            currSum += num;
            minSum = Math.min(minSum, currSum);
        }
        return 1 - minSum;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] nums = new int[n];

        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }
        
        Solution solution = new Solution();
        int result = solution.minStartValue(nums);
        System.out.println("The minimum start value is: " + result);
        
        scanner.close();
    }
}
